PlaybackLab/FullscreenSampleApp/app/src/main/java/com/ooyala/fullscreensampleapp/PlayerActivity.java

```diff
@@ -35,20 +36,33 @@
     private List<Data> dataList;
 
     private class ScrollListener extends RecyclerView.OnScrollListener {
-        private int snapPosition = RecyclerView.NO_POSITION;
-
-        @Override
-        public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newState) {
-            super.onScrollStateChanged(recyclerView, newState);
+        private static final int PLAY_DELAY = 500;
 
-            if (newState == RecyclerView.SCROLL_STATE_IDLE) {
+        private int snapPosition = RecyclerView.NO_POSITION;
+        private int state = RecyclerView.SCROLL_STATE_IDLE;
+        private Handler handler = new Handler();
+        private Runnable runnable = () -> {
+            if (state == RecyclerView.SCROLL_STATE_IDLE) {
                 int currentPosition = getCurrentPosition();
                 if (snapPosition != currentPosition) {
                     snapPosition = currentPosition;
                     initItem(snapPosition);
                 }
                 play(snapPosition);
             }
+        };
+
+        @Override
+        public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newState) {
+            super.onScrollStateChanged(recyclerView, newState);
+
+            state = newState;
+
+            //As the user scrolls, the video autoplays when the player is fully in view AND
+            // the scrolling pauses for 500 ms
```

Comment:
And I cannot understand where you pause scrolling for 500ms?

Reply:
`SCROLL_STATE_IDLE` becomes after `SCROLL_STATE_DRAGGING` or `SCROLL_STATE_SETTLING` and the delay in 500 ms added and if the state is also `SCROLL_STATE_IDLE` the video starts auto-playing. In other words, the comment means that the scroll is in a pause for 500 ms - it is a reason to start the playback